mne/io/brainvision/brainvision.py

```diff
@@ -313,57 +317,111 @@ def _read_vmrk_events(fname, event_id=None, trig_shift_by_type=None):
 
     # extract event information
     items = re.findall(r"^Mk\d+=(.*)", mk_txt, re.MULTILINE)
-    events, dropped = list(), list()
+    onset, duration, description = list(), list(), list()
+    date_str = None
     for info in items:
-        mtype, mdesc, onset, duration = info.split(',')[:4]
-        onset = int(onset) - 1  # BrainVision is 1-indexed, not 0-indexed
-        duration = (int(duration) if duration.isdigit() else 1)
-        if mdesc in event_id:
-            trigger = event_id[mdesc]
+        mtype, mdesc, this_onset, this_duration = info.split(',')[:4]
+        if mtype == 'New Segment':
+            date_str = info.split(',')[-1]
         else:
-            try:
-                # Match any three digit marker value (padded with whitespace).
-                # In BrainVision Recorder, the markers sometimes have a prefix
-                # depending on the type, e.g., Stimulus=S, Response=R,
-                # Optical=O, ... Note that any arbitrary stimulus type can be
-                # defined. So we match any single character that is not
-                # forbidden by BrainVision Recorder: [^a-z$%\-@/\\|;,:.\s]
-                marker_regexp = r'^[^a-z$%\-@/\\|;,:.\s]{0,1}([\s\d]{2}\d{1})$'
-                trigger = int(re.findall(marker_regexp, mdesc)[0])
-            except IndexError:
-                trigger = None
-        if mtype.lower() in trig_shift_by_type:
-            cur_shift = trig_shift_by_type[mtype.lower()]
-            if cur_shift is not None:
-                trigger += cur_shift
-            else:
-                # The trigger has been deliberately shifted to None. Do not
-                # add this to "dropped" so we do not warn about something
-                # that was done deliberately. Just continue with next item.
-                trigger = None
-                continue
-        # FIXME: ideally, we would not use the middle column of the events
-        # array to store the duration. A better solution would be using
-        # annotations.
-        if trigger:
-            events.append((onset, duration, trigger))
+            this_onset = int(this_onset) - 1  # BV is 1-indexed, not 0-indexed
+            this_duration = (int(this_duration)
+                             if this_duration.isdigit() else 1)
+            duration.append(this_duration)
+            onset.append(this_onset)
+            description.append(mtype + '/' + mdesc)
+
+    return np.array(onset), np.array(duration), np.array(description), date_str
+
+
+def _event_id_func(desc, event_id, trig_shift_by_type):
+    # extract event information
```

Comment:
Not our doctoring format

Reply:
to me the problem is not the docstring, but the function. If instead of `_event_id_func` we call it `_extract_event_information_blabla` there's no need for the docstring

Reply:
Then `event_id_func=_event_id_func` becomes `event_id_func=_extract_event_information_bla`.